### PR TITLE
onionshare: 2.4 -> 2.5

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
https://github.com/onionshare/onionshare/releases/tag/v2.5

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### TODO
- [ ] ~package `meek`~
- [x] package `snowflake`
- [x] package `cepa` properly

### CODE REVIEW

**Comment by lourkeur** [comment]
postponed `meek` because I'm stuck and Onionshare 2.5 without meek is still better than 2.4.

File: pkgs/tools/networking/snowflake/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    url = meta.repositories.git;
+    rev = "v${version}";
+    hash = "sha256-ULkqsh0DeFI1GsaVaHGSjoEY38EktvDVC52Sx6cQLOE=";
+  };
+
+  vendorSha256 = "D5A19UHL1WEE1ODT80jKT+PJ5CTXPjc9Eg6v2Nfm4aw=";
+
+  meta = with lib; {
+    maintainers = with maintainers; [ lourkeur ];
+    repositories.git = "https://git.torproject.org/pluggable-transports/snowflake.git";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:16]:
```suggestion
```
We do not set this anymore.

File: pkgs/tools/networking/snowflake/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "snowflake";
+  version = "2.0.1";
+  src = fetchgit {
+    url = meta.repositories.git;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:7]:
```suggestion
    url = "https://git.torproject.org/pluggable-transports/snowflake.git";
```

File: pkgs/tools/networking/snowflake/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{ lib, buildGoModule, fetchgit }:
+
+buildGoModule rec {
+  pname = "snowflake";
+  version = "2.0.1";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:6]:
```suggestion
  version = "2.0.1";

  src = fetchgit {
```

File: pkgs/tools/networking/snowflake/default.nix [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    url = meta.repositories.git;
+    rev = "v${version}";
+    hash = "sha256-ULkqsh0DeFI1GsaVaHGSjoEY38EktvDVC52Sx6cQLOE=";
+  };
+
+  vendorSha256 = "D5A19UHL1WEE1ODT80jKT+PJ5CTXPjc9Eg6v2Nfm4aw=";
+
+  meta = with lib; {
+    maintainers = with maintainers; [ lourkeur ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/snowflake/default.nix:15]:
```suggestion
    description = "System to defeat internet censorship";
    homepage = "https://snowflake.torproject.org/";
    maintainers = with maintainers; [ lourkeur ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by risicle** [comment]
https://nvd.nist.gov/vuln/detail/CVE-2022-21688
https://nvd.nist.gov/vuln/detail/CVE-2022-21689
https://nvd.nist.gov/vuln/detail/CVE-2022-21690
https://nvd.nist.gov/vuln/detail/CVE-2022-21691
https://nvd.nist.gov/vuln/detail/CVE-2022-21692
https://nvd.nist.gov/vuln/detail/CVE-2022-21693
https://nvd.nist.gov/vuln/detail/CVE-2022-21694
https://nvd.nist.gov/vuln/detail/CVE-2022-21695
https://nvd.nist.gov/vuln/detail/CVE-2022-21696

**Pull request merged**